subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/ResolveTaskExecutionModeExecuter.java

```diff
@@ -94,7 +94,7 @@ public String getDisplayName() {
             return executer.execute(task, state, context);
         } finally {
             outputs.setPreviousOutputFiles(null);
-            context.setTaskArtifactState(null);
+            context.setTaskExecutionMode(null);
```

Comment:
Why do we sometimes clear things from the context (like here) and don't in other places (like `ResolveBeforeExecutionStateTaskExecuter`)?

Reply:
Maybe we do the cleaning so we ensure no one up the chain accesses task artifact state?

Now that the arguments of the setters are marked as non null, we also get warnings here about passing in `null`. Shall we remove the clearing?

Reply:
This is not a regression in this PR. I also intend to completely revamp this when moving to `:execution` in a way that makes access to unavailable data impossible. I don't want to touch this stuff until then.